### DISCUSSION TITLE
Support subfolders for step definition files

### DISCUSSION THREAD

**Discussion by sbedi123** [behave/behave#977] [Q&A] [answered]
Does behave support subfolders for step definition files. Can I have subfolders inside the steps directory which will contain the steps implementation in behave?

Please advice

Thanks
SB

**Comment by jenisys** [comment, marked as answer]
Currently not.
If you have enough complexity to need subfolders, you should consider to use step-libraries.

ADVANTAGES:
* You are able to share the steps between different projects
* You are able to test your step-libraries (and ensure that they are working; especially. if you need to modify/adapt them) before you are using them in feature files

HINT:
That is basically them same approach that you normally use for the **test infrastructure / test automation** code.
Such kind of code makes it simpler for you to write test / automation needed in tests.

SEE ALSO:
* [features/step.use_step_library.feature](https://github.com/behave/behave/blob/master/features/step.use_step_library.feature)
